action-scheduler.php

```diff
@@ -26,27 +26,29 @@
  * @package ActionScheduler
  */
 
-if ( ! function_exists( 'action_scheduler_register_3_dot_5_dot_2' ) && function_exists( 'add_action' ) ) {
+if ( ! function_exists( 'action_scheduler_register_3_dot_5_dot_2' ) && function_exists( 'add_action' ) ) { // WRCS: DEFINED_VERSION.
+
+	define( 'ACTION_SCHEDULER_VERSION', '3.5.2' ); // WRCS: DEFINED_VERSION.
```

Comment:
Projecting forward a little, suppose we have Plugin A (loads first) with Action Scheduler 3.5.2 and Plugin B (loads second) with Action Scheduler 3.5.3:

- Plugin A will load Action Scheduler first.
- Function `action_scheduler_register_3_dot_5_dot_2` does not exist, so we enter this block.
- We define `ACTION_SCHEDULER_VERSION`.
- Later in the request, Plugin B loads its copy of Action Scheduler.
- Function `action_scheduler_register_3_dot_5_dot_3` does not exist, so we enter Plugin B's copy of this block.
- We attempt to define `ACTION_SCHEDULER_VERSION` a second time:
    - This leads to a notice, warning or error (depending on PHP version).
    - The constant will still be set to the initial value (`3.5.2` in this scenario), which will make the call to `$versions->register()` incorrect, in the case of Plugin B.

Reply:
Fair point. Perhaps add validation? `if ( ! defined( 'ACTION_SCHEDULER_VERSION' ) ) {...}`

Reply:
That would guard against the constant-already-defined notice/warning/error, but I *think* it would still lead to problems, because if an early version of Action Scheduler is loaded first then that is the version we capture in the constant (and so when a more recent version of Action Scheduler is also present, but loads second, we'd have essentially the same problem in that it registers itself as an earlier version than it really is).